### PR TITLE
Hardhat-ethers with legacy ethers v4

### PR DESCRIPTION
**Pull request by scnale**
NOTE: This pull request shouldn't be merged. It is only for comparing and testing purposes.

This tweaks `hardhat-ethers` to be compatible with `ethers` v4 and sets both `ethers` and `ethereum-waffle` dependencies to appropriate versions.

What's missing: choosing a legacy version number for both `hardhat-ethers` and `hardhat-waffle`.

### CODE REVIEW

File: packages/hardhat-waffle/package.json [hunk 1]
```diff
@@ -33,14 +33,14 @@
...
+    "ethereum-waffle": "^2.3.0",
+    "ethers": "^4.0.0",
     "hardhat": "^0.1.0-rc.0"
   },
   "peerDependencies": {
     "@nomiclabs/hardhat-ethers": "^0.1.0-rc.0",
-    "ethers": "^5.0.0",
-    "ethereum-waffle": "^3.0.0",
+    "ethers": "^4.0.0",
+    "ethereum-waffle": "^2.3.0",
```
Comment (alcuadrado) [inline packages/hardhat-waffle/package.json:43]: This plugin has a lot of imported things from waffle. Did you check that they are in sync with this version?

Comment (alcuadrado) [reply packages/hardhat-waffle/package.json:43]: By imported I mean copy-pasted code.

Comment (scnale) [reply packages/hardhat-waffle/package.json:43]: Hm, let me check.

**Comment by fvictorio** [comment]
The ethers and waffle plugins should have version `1.0.0`, right?

**Comment by alcuadrado** [comment]
> The ethers and waffle plugins should have version `1.0.0`, right?

Yes, that's right, @fvictorio

**Comment by alcuadrado** [comment]
Manually merged into `hardhart-ethersv4-plugins`.

**Pull request closed without merging**